firmware/common/portapack_persistent_memory.cpp

```diff
@@ -923,5 +923,81 @@ size_t data_size() {
     return sizeof(data_t);
 }
 
+bool dump_persistent_memory() {
+    std::string debug_dir = "DEBUG";
+    make_new_directory(debug_dir);
+
+    std::filesystem::path filename{};
+    File pmem_dump_file{};
+    filename = next_filename_matching_pattern(debug_dir + "/PMEMDUMP_????.*");
+    if (filename.empty()) {
+        return false;
+    }
+    filename = filename.replace_extension(u".TXT");
+    auto result = pmem_dump_file.create(filename);  // First freq if no output file
+    if (!result.is_valid()) {
+        std::string strbuf{};
+
+        strbuf = "structure_version: " + to_string_dec_uint(data->structure_version);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "target_frequency: " + to_string_dec_int(data->target_frequency);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "correction_ppb: " + to_string_dec_int(data->correction_ppb);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "modem_def_index: " + to_string_dec_uint(data->modem_def_index);
+        pmem_dump_file.write_line(strbuf);
+        // strbuf = "serial_format: " + to_string_dec_uint( data -> serial_format );
+        // pmem_dump_file.write_line(strbuf);
+        strbuf = "modem_bw: " + to_string_dec_int(data->modem_bw);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "afsk_mark_freq: " + to_string_dec_int(data->afsk_mark_freq);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "afsk_space_freq: " + to_string_dec_int(data->afsk_space_freq);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "modem_baudrate: " + to_string_dec_int(data->modem_baudrate);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "modem_repeat: " + to_string_dec_int(data->modem_repeat);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "playdead_magic: " + to_string_dec_uint(data->playdead_magic);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "playing_dead: " + to_string_dec_uint(data->playing_dead);
+        pmem_dump_file.write_line(strbuf);
+        strbuf = "playdead_sequence: " + to_string_dec_uint(data->playdead_sequence);
+        pmem_dump_file.write_line(strbuf);
+        // strbuf = "ui_config: " + to_string_dec_uint( data -> ui_config );
```

Comment:
IMHO, I would propose just dumping out these bitmaps in hex (vs not showing them)...

Reply:
Nah, i left them yesterday because I had no more time.
I'm going to dump each bit correctly using the method provided.